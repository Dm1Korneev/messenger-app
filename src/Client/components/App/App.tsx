import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';

import { useIsLoggedIn } from 'Client/hooks';

import { MessengerScreen } from '../MessengerScreen';
import { SignIn } from '../SignIn';

export const App = () => {
  const isLoggedIn = useIsLoggedIn();

  return (
    <Box display="flex" height="100vh">
      <CssBaseline />
      {isLoggedIn ? <MessengerScreen /> : <SignIn />}
    </Box>
  );
};
