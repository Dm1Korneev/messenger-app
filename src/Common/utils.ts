
type Filter = Parameters<typeof Array.prototype.filter>[0]
export const distinct: Filter = (value, index, self) => self.indexOf(value) === index;
