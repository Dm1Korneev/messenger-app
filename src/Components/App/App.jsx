import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import CssBaseline from '@material-ui/core/CssBaseline';

import MessengerScreen from 'Components/MessengerScreen';
import SignIn from 'Components/SignIn';

const App = ({
  isLoggedIn,
  loginFromStore,
}) => {
  useEffect(() => {
    loginFromStore();
  }, [loginFromStore]);

  let result;
  if (!isLoggedIn) {
    result = <SignIn />;
  } else {
    result = (
      <MessengerScreen />
    );
  }

  return (
    <Box height="100vh" display="flex">
      <CssBaseline />
      {result}
    </Box>
  );
};

App.propTypes = {
  loginFromStore: PropTypes.func.isRequired,
  isLoggedIn: PropTypes.bool.isRequired,
};

export default App;
