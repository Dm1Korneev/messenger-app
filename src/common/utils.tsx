export function distinct(value: any, index: number, self: Array<any>): boolean {
  return self.indexOf(value) === index;
}
