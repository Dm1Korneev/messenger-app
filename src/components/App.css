.App__content {
  max-width: 1010px;
  min-width: 300px;
  margin: 0 auto;
  padding: 1em;
  background: #fff;
  border-radius: 0.5em;

  height: 90vh;
  display: flex;
  flex-flow: column;
  justify-content: space-between;
}
