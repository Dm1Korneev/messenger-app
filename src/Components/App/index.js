import commonHoc from 'Components/commonHoc';

import { loginFromStore } from 'Redux/actions';
import { chatDialogIsOpenSelector, isLoggedInSelector, userModifyDialogIsOpenSelector } from 'Selectors/session';

import App from './App';

const mapStateToProps = (state) => ({
  chatDialogIsOpen: chatDialogIsOpenSelector(state),
  userModifyDialogIsOpen: userModifyDialogIsOpenSelector(state),
  isLoggedIn: isLoggedInSelector(state),
});

const mapDispatchToProps = {
  loginFromStore,
};

const Container = commonHoc(App, {
  mapStateToProps,
  mapDispatchToProps,
});

export { Container as App };
