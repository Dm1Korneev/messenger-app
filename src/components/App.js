import React, { Component } from "react";

// components
import MessagesList from "./MessagesList";
import MessageInput from "./MessageInput";
import SignIn from "./SignIn";

// @material-ui
import withStyles from "@material-ui/core/styles/withStyles";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import { blue } from "@material-ui/core/colors";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import Button from "@material-ui/core/Button";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";

// common
import { getMessages, sendMessage } from "../common/messengerAPI";
import {
  isLoggedIn,
  getToken,
  currentUser,
  login,
  register,
  saveToken,
  logout
} from "../common/authentication";

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      messages: [],
      token: getToken(),
      currentUser: currentUser()
    };

    this.theme = createMuiTheme({
      palette: {
        primary: blue
      },
      typography: {
        useNextVariants: true
      }
    });
  }

  componentDidMount() {
    const { token } = this.state;
    if (isLoggedIn(token)) {
      this.loadMessages();
    }
  }

  loadMessages = () => {
    const { token } = this.state;

    getMessages(token, messages => this.setState({ messages: messages }));
  };

  onSendMessage = messageText => {
    const { token } = this.state;

    sendMessage(token, messageText);
    this.loadMessages();
  };

  handleClickLogout = () => {
    logout();
    this.setState({
      token: undefined
    });
  };

  onSignIn = ({ email, password, remember }) => {
    login(email, password, result => {
      const { token } = result;

      if (remember) {
        saveToken(token);
      }
      this.setState({
        token: token
      });
      this.loadMessages();
    });
  };

  onRegister = ({ email, password, name, remember }) => {
    register(email, password, name, result => {
      const { token } = result;

      if (remember) {
        saveToken(token);
      }
      this.setState({
        token: token
      });
      this.loadMessages();
    });
  };

  render() {
    const { classes } = this.props;
    const { messages, token } = this.state;
    const user = currentUser(token);

    let result;
    if (!isLoggedIn(token)) {
      result = <SignIn onSignIn={this.onSignIn} onRegister={this.onRegister} />;
    } else {
      result = (
        <>
          <AppBar position="static">
            <Toolbar variant="dense">
              <IconButton
                className={classes.menuButton}
                color="inherit"
                aria-label="Menu"
              >
                <MenuIcon />
              </IconButton>
              <Typography variant="h6" color="inherit" className={classes.grow}>
                {user.name}
              </Typography>
              <Button color="inherit" onClick={this.handleClickLogout}>
                logout
              </Button>
            </Toolbar>
          </AppBar>
          <Grid item xs={12} sm={11} md={9} lg={7} className={classes.grid}>
            <Paper className={classes.content}>
              <MessagesList messages={messages} />
              <MessageInput onSendMessage={this.onSendMessage} />
            </Paper>
          </Grid>
        </>
      );
    }

    return (
      <main className={classes.root}>
        <MuiThemeProvider theme={this.theme}>{result}</MuiThemeProvider>
      </main>
    );
  }
}

const styles = theme => ({
  root: {
    height: "100vh"
  },
  grid: {
    margin: "0 auto",
    height: "calc(100% - 48px)",
    padding: theme.spacing.unit * 2
  },
  content: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    paddingLeft: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingTop: theme.spacing.unit
  },
  grow: {
    flexGrow: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  }
});

export default withStyles(styles)(App);
