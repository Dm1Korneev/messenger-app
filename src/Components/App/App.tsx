import Box from '@material-ui/core/Box';
import CssBaseline from '@material-ui/core/CssBaseline';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';

import { MessengerScreen } from 'Components/MessengerScreen';
import { SignIn } from 'Components/SignIn';
import { useIsLoggedIn } from 'Hooks';
import { loginFromStore } from 'Redux/actions';

export const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loginFromStore());
  }, [dispatch]);

  const isLoggedIn = useIsLoggedIn();

  return (
    <Box height="100vh" display="flex">
      <CssBaseline />
      {isLoggedIn ? <MessengerScreen /> : <SignIn />}
    </Box>
  );
};
