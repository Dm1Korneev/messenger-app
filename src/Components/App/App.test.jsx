import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { render } from '@testing-library/react';

import { mockComponent } from 'testing/utils';

import App from './App';

const emptyStore = createStore(() => { });

jest.mock('Components/TopBar', () => mockComponent('Icon'));
jest.mock('Components/SignIn', () => mockComponent('SignIn'));
jest.mock('Components/SideBar', () => mockComponent('SideBar'));
jest.mock('Components/TopBar', () => mockComponent('TopBar'));
jest.mock('Components/MainContent', () => mockComponent('MainContent'));
jest.mock('@material-ui/core/CssBaseline', () => mockComponent('CssBaseline'));

describe('render app when not logged', () => {
  const loginFromStore = jest.fn();

  const props = {
    loginFromStore,
    chatDialogIsOpen: false,
    userModifyDialogIsOpen: false,
    isLoggedIn: false,
  };

  const setup = () => render(
    <Provider store={emptyStore}>
      <App {...props} />
    </Provider>,
  );

  test('loginFromStore function is call', async () => {
    setup();

    expect(loginFromStore).toHaveBeenCalled();
  });

  test('snapshot test', async () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
