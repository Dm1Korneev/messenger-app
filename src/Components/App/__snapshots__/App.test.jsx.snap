// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app when not logged snapshot test 1`] = `
<DocumentFragment>
  <div
    data-originalcomponent="Box"
    test-prop-display="\\"flex\\""
    test-prop-height="\\"100vh\\""
  >
    <div
      data-originalcomponent="CssBaseline"
    />
    <div
      data-originalcomponent="SignIn"
    />
  </div>
</DocumentFragment>
`;
